### PR TITLE
feat: add loot criteria to extra metadata

### LINKED ISSUE #629: [Loot] `extra.items` JSON includes irrelevant loot, ignoring denylist
### Checklist

- [X] I'm reporting a problem with Dink
- [X] I've searched the [issues and pull requests](https://github.com/pajlads/Dinkplugin/issues?q=) for similar looking reports

### Describe your issue

Unsure if this is truly a bug — I assume at this point it's not but I'd like to at least understand why and potentially find a solution — but when receiving the payload JSON from a `LOOT` trigger, the `extra.items` array includes items irrelevant to the trigger if the drop included multiple items. This is true even if an item is on the denylist. 
These items are not included in the `embed` discord message, as expected,  so why are they not also filtered out of the `extra.items` content?

For example, I have an Allowlist of `Raw chicken` and kill a Chicken. The Discord message correctly shows that I got `1 x Raw Chicken`, while the JSON tells me I got [`1 x Bones`, `1 x Raw Chicken`]. I don't care about the bones, and this actually makes it a headache for me to process this data as I now need to keep a local Allowlist of items and check against it before processing the item data further. The Bones still show up in `extra.items` even if my Denylist includes `Bones`.

For context, I'm using Dink for a clan bingo event to create a list of Loot drops (in a Google Sheet, which is then processed via Google Apps Script to record, calculate and display tile progress, points, etc). There'll be hundreds of items needed from sources all across the game so even if the Denylist was respected that would not be a practical solution for me as I'd need to include every single other possible drop from the monsters who drop the items which _are_ wanted. The other solution would be to manually parse the `embed`, as this is already filtered, but that would require mandating a standard format for the message and seems sort of counterproductive when Dink already sends the JSON.

### Screenshots

image

image

**See `extra.items[0]`:**

image

### Runelite Logs

N/A

### Runelite Version

1.10.46

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #629

this change incurs a minor performance penalty since we now check the allowlist even for items that pass value/rarity threshold

### CODE REVIEW

File: src/main/java/dinkplugin/domain/LootCriteria.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package dinkplugin.domain;
+
+public enum LootCriteria {
+    ALLOWLIST,
+    DENYLIST,
+    VALUE,
+    RARITY
```
Comment (iProdigy) [inline src/main/java/dinkplugin/domain/LootCriteria.java:7]: should these be renamed to `VALUABLE`/RARE` ?`

Comment (pajlada) [reply src/main/java/dinkplugin/domain/LootCriteria.java:7]: nah I think this form is good

**Review by pajlada — APPROVED** [review]
Not tested in-game

**Review by Felanbird — APPROVED** [review]
only minimally tested, seems fine
```
  "extra":{
      "items":[
         {
            "criteria":[
               "DENYLIST",
               "VALUE"
            ],
            "id":526,
            "quantity":1,
            "priceEach":62,
            "name":"Bones"
         },
         {
            "rarity":0.02343566908835247,
            "criteria":[
               "VALUE"
            ],
            "id":1351,
            "quantity":1,
            "priceEach":88,
            "name":"Bronze axe"
         }
      ],
```

**Pull request merged**